controllers/storagecluster/cephcluster.go

```diff
@@ -1399,3 +1407,17 @@ func determineDefaultCephDeviceClass(foundDeviceClasses []rookCephv1.DeviceClass
 	// if no device classes are found in status return empty string
 	return ""
 }
+
+// isEncrptionSettingUpdated checks whether ecryption was enabled or disabled for the storageClassDeviceSet.
+func isEncrptionSettingUpdated(newDeviceSet, existingDeviceSet []rookCephv1.StorageClassDeviceSet) bool {
+	if len(newDeviceSet) != len(existingDeviceSet) {
+		return false
```

Comment:
I believe this is an unknown state instead of false, Is sending false safe here?

Reply:
This will happen in case of cluster expansion when the storagedeviceclass sets won't of the same size in the new and existing cluster.  Sending false should be safe here, as in the next reconcile the lengths would match.

Reply:
Will the list be in the status of the CR in the same order as spec?

Reply:
Good question. I don't think its guaranteed and code should not reply on this that.  I've updated the changes. Looks like we don't need to compare both the new and old storageclassdeviceSets at all.  Just comparing the new `clusterWide` encryption setting with the existing StorageClassDeviceSet would be enough. Let me know what you think.